osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -141,10 +144,18 @@ protected override void LoadComplete()
 
             protected abstract string FragmentShader { get; }
 
-            protected float GetSize()
+            public float GetSize()
             {
                 float size = defaultFlashlightSize * sizeMultiplier;
 
+                if (GetPlayfieldScale != null)
+                {
+                    Vector2 playfieldScale = GetPlayfieldScale();
+                    float rulesetScaleAvg = (playfieldScale.X + playfieldScale.Y) / 2f;
+
+                    size *= rulesetScaleAvg;
```

Comment:
I assume an X/Y average would be okay for this, even then AFAIK nowhere is there scaling applied with different X/Y changes.

Reply:
I've removed the average assumption in favour of an explicit assertion in https://github.com/ppy/osu/pull/26181/commits/115d82664bb0a3271e7593dacce987557a2ff098 (and then just assuming proportional scaling and taking the absolute of X).

Rationale being that stuff is being done elsewhere already like

https://github.com/ppy/osu/blob/b0c45c87c665361e22aafc9f1a01f7aeb5e7b053/osu.Game.Rulesets.Catch/Mods/CatchModFloatingFruits.cs#L23

The above didn't cause critical failure with the average, but only because scale is being applied to the playfield _adjustment container_ rather than the playfield.